Client/core/CChat.cpp

```diff
@@ -548,6 +550,31 @@ void CChat::ScrollDown()
     }
 }
 
+void CChat::SelectInputHistoryEntry(uint uiEntry)
+{
+    uint uiPreviouslySelectedInputHistoryEntry = m_uiSelectedInputHistoryEntry;
+
+    // Check if we're in bounds, otherwise clear selection
+    if (!m_InputHistory.vecEntryList.empty() && uiEntry > 0 && uiEntry < m_InputHistory.vecEntryList.size())

```

Comment:
Maybe could implement `empty` and `size` on `SInputHistory` instead of having to access vecEntryList manually each time? i.e eventually `vecEntryList` could be made private. Not sure what your plans are for this though, so it's up to you

Reply:
Yup, I actually made those initially but I started having doubts about if they are good or bad. I think I should just add them in for now.